### ISSUE TITLE
rule `prefer-string-replace-all` enforces invalid syntax

### ISSUE THREAD

**Issue by notramo** [oxc-project/oxc#1790]
`eslint-plugin-unicorn(prefer-string-replace-all)` warns on the following:
```javascript
"Hello world".replaceAll(/world/g, 'world!');
```
It is already a `replaceAll` call, but I guess it only checks that the regex is global. However, `replaceAll` must be called with a global regex.

**Comment by Boshen** [comment]
Thank you for the bug report.

It seems like a broken test case from our end https://github.com/sindresorhus/eslint-plugin-unicorn/blob/ce44929dbdca42222407a4e19d0c17e1be748ec3/test/prefer-string-replace-all.mjs#L109
